### ISSUE TITLE
[BUG] DataDog trace hide the php errors in PHP 7.4

### ISSUE THREAD

**Issue by denis-bonci** [DataDog/dd-trace-php#2049]
### Bug description
I noticed on Data Dog some logs contains error like this `Notice: Undefined variable: response`.

This error occurs because the variable [$response is null](https://github.com/denis-bonci/dd-tests/blob/f1a74223ab2ad9238ed88389c32037e2c1df5966/php-7.4_symfony-4.4/public/index.php#L26), but the "real" error instead is usually a PHP error, for instance [something like this](https://github.com/denis-bonci/dd-tests/blob/f1a74223ab2ad9238ed88389c32037e2c1df5966/php-7.4_symfony-4.4/src/Controller/DefaultController.php#L11-L12).

The flow is interrupted because the custom error handler \Symfony\Component\ErrorHandler\ErrorHandler::handleError (configure by Symfony with the set_error_handler function) is not called.

This weird flow occurs only when datadog tracer intercept Symfony internal calls using hooks like [this](https://github.com/DataDog/dd-trace-php/blob/d0d60c527451abf09dd06d2714166034f6280b9f/src/Integrations/Integrations/Symfony/SymfonyIntegration.php#L182-L184).

If instead the PHP error occurs in another part of the code where Data Dog has no hooks (for instance the index.php), this error is normally handled by symfony with \Symfony\Component\ErrorHandler\ErrorHandler::handleError.

This means that the DataDog tracer break the normal error handling flow in case of PHP error.

Finally, I configured locally different PHP versions with different Symfony versions, forcing a PHP error inside a controller, in order to understand if this error was fixed and it seems fixed from the PHP 8.x version but still exists in PHP 7.x.

The testing project is [available here](https://github.com/denis-bonci/dd-tests/).

**PHP 7.4
Symfony 4.4**
![image (4)](https://github.com/DataDog/dd-trace-php/assets/7313285/4e92bbf5-8c61-4292-848e-df63fcc946da)

**PHP 7.4
Symfony 5.4**
![image (5)](https://github.com/DataDog/dd-trace-php/assets/7313285/8d0e30ba-7331-4d76-a7d2-94e5a3a71c70)

**PHP 8.2
Symfony 4.4**
![image (6)](https://github.com/DataDog/dd-trace-php/assets/7313285/89c6d400-2462-45f0-9e09-9c5160788574)

**PHP 8.2
Symfony 6.2**
![image (7)](https://github.com/DataDog/dd-trace-php/assets/7313285/ef6dba1d-55b6-4e8b-b2fa-a79782e86e7a)

### PHP version
PHP 7.4.33 (cli) (built: Nov 12 2022 09:17:36) ( NTS )
Copyright (c) The PHP Group
Zend Engine v3.4.0, Copyright (c) Zend Technologies
    with ddtrace v0.86.3, Copyright Datadog, by Datadog
    with datadog-profiling v0.15.1, Copyright Datadog, by Datadog
    with ddappsec v0.7.2, Copyright Datadog, by Datadog

### Tracer version
0.86.3

### Installed extensions
[PHP Modules]
Core
ctype
curl
datadog-profiling
date
ddappsec
ddtrace
dom
fileinfo
filter
ftp
hash
iconv
json
libxml
mbstring
mysqlnd
openssl
pcre
PDO
pdo_sqlite
Phar
posix
readline
Reflection
session
SimpleXML
sodium
SPL
sqlite3
standard
tokenizer
xml
xmlreader
xmlwriter
zlib

[Zend Modules]
datadog-profiling
ddappsec
ddtrace

### OS info
NAME="Alpine Linux"
VERSION_ID=3.16.3
PRETTY_NAME="Alpine Linux v3.16"

#### Output of phpinfo() (ddtrace >= 0.47.0)
{ "date": "2023-05-12T11:09:34Z", "os_name": "Linux 5ac58e11c0a8 5.15.0-71-generic #78~20.04.1-Ubuntu SMP Wed Apr 19 11:26:48 UTC 2023 x86_64", "os_version": "5.15.0-71-generic", "version": "0.86.3", "lang": "php", "lang_version": "7.4.33", "env": null, "enabled": true, "service": null, "enabled_cli": false, "agent_url": "http:\/\/localhost:8126", "debug": false, "analytics_enabled": false, "sample_rate": 1, "sampling_rules": [], "tags": [], "service_mapping": [], "distributed_tracing_enabled": true, "priority_sampling_enabled": true, "dd_version": null, "architecture": "x86_64", "sapi": "cgi-fcgi", "datadog.trace.request_init_hook": "\/opt\/datadog\/dd-library\/0.86.3\/dd-trace-sources\/bridge\/dd_wrap_autoloader.php", "open_basedir_configured": false, "uri_fragment_regex": null, "uri_mapping_incoming": null, "uri_mapping_outgoing": null, "auto_flush_enabled": false, "generate_root_span": true, "http_client_split_by_domain": false, "measure_compile_time": true, "report_hostname_on_root_span": false, "traced_internal_functions": null, "auto_prepend_file_configured": false, "integrations_disabled": "default", "enabled_from_env": true, "opcache.file_cache": null, "agent_error": "Failed to connect to localhost port 8126 after 0 ms: Connection refused" } 

![Screenshot from 2023-05-12 14-33-00](https://github.com/DataDog/dd-trace-php/assets/7313285/4991eb97-6493-4e1b-96de-1eb1e81a4512)

**Comment by PROFeNoM** [comment]
Hey Denis! Thanks for the throughout description of the issue, this is some really valuable information here 👍 Someone complained about a _very similar_ issue not so long ago (#1955). I'm on it and I'll to find what's going there 🔍

**Comment by PROFeNoM** [comment]
:wave:

Could you please tell me if you still have this issue with the newest release (0.87.0). So far, when testing on the testing project you've provided (thank you for that :+1:), this issue seems to have been resolved going from 0.86.3 to 0.87.0.

**Comment by denis-bonci** [comment]
Hi @PROFeNoM ,
thanks for your feedback.

I tried to rebuild the whole docker project and install the latest version, but the issue seems to still exists in PHP 7.4.

Indeed the error I'm expecting to receive is 

> Call to a member function hello() on null

and I received instead

> Call to a member function send() on null

```bash
PHP 7.4.33 (cli) (built: Nov 12 2022 09:17:36) ( NTS )
Copyright (c) The PHP Group
Zend Engine v3.4.0, Copyright (c) Zend Technologies
    with ddtrace v0.87.0, Copyright Datadog, by Datadog
    with datadog-profiling v0.87.0+dd01fad553ad51c758eb3ec64387d471dbd06be6, Copyright Datadog, by Datadog
    with ddappsec v0.8.0, Copyright Datadog, by Datadog
```

**Symfony 4.4**

![image](https://github.com/DataDog/dd-trace-php/assets/7313285/c4673907-cf2f-40d2-b369-661000eb5ee9)


**Symfony 5.4**

![image](https://github.com/DataDog/dd-trace-php/assets/7313285/05ad0a72-83ab-4d82-af0f-4e50843e7ebd)

This is with PHP 8.2

```bash
PHP 8.2.6 (cli) (built: May 11 2023 20:14:18) (NTS)
Copyright (c) The PHP Group
Zend Engine v4.2.6, Copyright (c) Zend Technologies
    with ddtrace v0.87.0, Copyright Datadog, by Datadog
    with datadog-profiling v0.87.0+dd01fad553ad51c758eb3ec64387d471dbd06be6, Copyright Datadog, by Datadog
    with ddappsec v0.8.0, Copyright Datadog, by Datadog
```

**Symfony 4.4**

![image](https://github.com/DataDog/dd-trace-php/assets/7313285/6f804ab0-3dcd-42f3-8646-fb17c81570e3)

**Symfony 6.2**

![image](https://github.com/DataDog/dd-trace-php/assets/7313285/93d3b6b1-dbf9-4fc9-8ed9-054b2b26fe62)

Unfortunately it seems not fixed yet.

Thanks.

**Comment by bwoebi** [comment]
Hey @denis-bonci,

we've just released 0.87.1 containing a fix for this - thanks for reporting!

**Comment by denis-bonci** [comment]
> Hey @denis-bonci,
> 
> we've just released 0.87.1 containing a fix for this - thanks for reporting!

Thanks to you for fixing the bug so fast.
